workflow/controller/operator.go

```diff
@@ -1822,6 +1822,18 @@ func (woc *wfOperationCtx) possiblyGetRetryChildNode(node *wfv1.NodeStatus) *wfv
 			if childNode == nil {
 				continue
 			}
+
+			// childNode.nodeFlag was missing, this can only happen if NodeStatus is old
+			if childNode.NodeFlag == nil {
+				// fallback where nodeFlags is absent
+				// ensure that we do not return
+				if !common.CheckHookNode(childNode.Name) {
+					return childNode
+				} else {
+					continue
+				}
+			}
+
 			if childNode.NodeFlag == nil || !childNode.NodeFlag.Hooked {
 				return childNode
 			}
```

Comment:
Can remove the nil check here, but this is very intentionally kept because the above code isn't meant to be around for long.